File: problems/1370_increasing_decreasing_string/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string sortString(string s) {
        int count[26] = {};
        for (char c : s) count[c-'a']++;
        string res;
        while ((int)res.size() < (int)s.size()) {
            for (int i = 0; i < 26; i++) if (count[i] > 0) { res += (char)(i+'a'); count[i]--; }
            for (int i = 25; i >= 0; i--) if (count[i] > 0) { res += (char)(i+'a'); count[i]--; }
        }
        return res;
    }
};

File: problems/1370_increasing_decreasing_string/Solution.java
class Solution {
    public String sortString(String s) {
        int[] count = new int[26];
        for (char c : s.toCharArray()) count[c-'a']++;
        StringBuilder res = new StringBuilder();
        while (res.length() < s.length()) {
            for (int i = 0; i < 26; i++) if (count[i] > 0) { res.append((char)(i+'a')); count[i]--; }
            for (int i = 25; i >= 0; i--) if (count[i] > 0) { res.append((char)(i+'a')); count[i]--; }
        }
        return res.toString();
    }
}
